### ISSUE TITLE
Unit tests broken on Go 1.21

### ISSUE THREAD

**Issue by honza** [metal3-io/cluster-api-provider-metal3#1431]
Running `make unit` produces the following errors:

```
Summarizing 9 Failures:
  [FAIL] Metal3Data manager Test Data Reconcile functions Test Reconcile [It] Deletion, Cluster not found
  /home/honza/code/metal3-io/cluster-api-provider-metal3/controllers/metal3data_controller.go:137
  [FAIL] Metal3Data manager Test Data Reconcile functions Test Reconcile [It] Deletion, release requeue
  /home/honza/code/metal3-io/cluster-api-provider-metal3/controllers/metal3data_controller.go:137
  [FAIL] Metal3Data manager Test Data Reconcile functions Test Reconcile [It] Deletion, release error
  /home/honza/code/metal3-io/cluster-api-provider-metal3/controllers/metal3data_controller.go:137
  [FAIL] Metal3Data manager Test Data Reconcile functions Test Reconcile [It] Reconcile normal error
  /home/honza/code/metal3-io/cluster-api-provider-metal3/controllers/metal3data_controller.go:127
  [FAIL] Metal3Data manager Test Data Reconcile functions Test Reconcile [It] Reconcile normal no error
  /home/honza/code/metal3-io/cluster-api-provider-metal3/controllers/metal3data_controller.go:127
  [FAIL] Metal3DataTemplate manager Test Reconcile [It] Deletion, Cluster not found
  /home/honza/code/metal3-io/cluster-api-provider-metal3/controllers/metal3datatemplate_controller.go:153
  [FAIL] Metal3DataTemplate manager Test Reconcile [It] Deletion, Cluster not found, error
  /home/honza/code/metal3-io/cluster-api-provider-metal3/controllers/metal3datatemplate_controller.go:153
  [FAIL] Metal3DataTemplate manager Test Reconcile [It] Reconcile normal error
  /home/honza/code/metal3-io/cluster-api-provider-metal3/controllers/metal3datatemplate_controller.go:143
  [FAIL] Metal3DataTemplate manager Test Reconcile [It] Reconcile normal no error
  /home/honza/code/metal3-io/cluster-api-provider-metal3/controllers/metal3datatemplate_controller.go:143

Ran 166 of 166 Specs in 8.403 seconds
FAIL! -- 157 Passed | 9 Failed | 0 Pending | 0 Skipped
--- FAIL: TestControllers (8.45s)
FAIL
coverage: 80.1% of statements
FAIL	github.com/metal3-io/cluster-api-provider-metal3/controllers	8.670s
ok  	github.com/metal3-io/cluster-api-provider-metal3/baremetal	10.472s	coverage: 82.7% of statements
ok  	github.com/metal3-io/cluster-api-provider-metal3/baremetal/remote	0.056s	coverage: 100.0% of statements
FAIL
make: *** [Makefile:115: unit] Error 1
```

**Comment by tuminoid** [comment]
This has been fixed by bumping Go  to 1.21 last week, also fixing the issues with it.
https://github.com/metal3-io/cluster-api-provider-metal3/pull/1435

**Comment by Rozzii** [comment]
> This has been fixed by bumping Go to 1.21 last week, also fixing the issues with it. #1435

Then I am closing this, thanks @tuminoid
